src/app/components/Pagination/index.tsx

```diff
@@ -0,0 +1,203 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react';
+import { useContext } from 'react';
+import {
+  GEL_GROUP_2_SCREEN_WIDTH_MIN,
+  GEL_GROUP_2_SCREEN_WIDTH_MAX,
+  GEL_GROUP_3_SCREEN_WIDTH_MAX,
+  GEL_GROUP_3_SCREEN_WIDTH_MIN,
+  GEL_GROUP_4_SCREEN_WIDTH_MIN,
+} from '#app/legacy/psammead/gel-foundations/src/breakpoints';
+import { RequestContext } from '#app/contexts/RequestContext';
+import { LIVE_PAGE } from '#app/routes/utils/pageTypes';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import buildBlocks, { VISIBILITY } from './buildBlocks';
+import { Ellipsis, LeftChevron, RightChevron } from '../icons';
+import VisuallyHiddenText from '../VisuallyHiddenText';
+import styles from './index.styles';
+
+interface LinkComponentProps {
+  children: number | JSX.Element;
```

Comment:
For instances where `children` are passed as props, you can use the built-in React wrapper type called `PropsWithChildren`, which will automatically type `children` for you.

For example in this case on line 70 you would write:
```
}: PropsWithChildren<LinkComponentProps>) => (
```

This will let you remove the manual declaration of `children` from your custom type.